### PR TITLE
Move locust config to central config file

### PR DESCRIPTION
**Pull request by heoelri**
This was brought up in yesterday's LevelUp session - the embedded locust config is currently burried in our pipeline code. This PR will change that and move the embedded locust config over to our central config file in `.ado/pipelines/config`.

This will close issue #151

### CODE REVIEW

**Comment by heoelri** [comment]
Issues when using `number` parameters in ADO templates. https://developercommunity.visualstudio.com/t/parsing-error-when-using-variable-as-parameter-val/1214362

**Comment by heoelri** [comment]
Testing in E2e: https://dev.azure.com/alwayson/Foundational-Online/_build/results?buildId=441&view=results

**Comment by heoelri** [comment]
Looks promising so far: 
image

**Comment by heoelri** [comment]
@sebader shouldn't be related to my changes here, but i noticed that the embedded load test failed:

image

The failure count is much higher than expected.

**Comment by heoelri** [comment]
@sebader i think this PR is good to go and can be merged - it was successfully validated here: https://dev.azure.com/alwayson/Foundational-Online/_build/results?buildId=441&view=results But the issue that the load test results were above our baseline values still remains. I'm running another embedded load test in main now to check if this was a transient issue or not.

**Comment by heoelri** [comment]
Same issue in `main` - will dig a bit deeper and will try to figure out what's causing the high error rate.

image
https://dev.azure.com/alwayson/Foundational-Online/_build/results?buildId=442&view=logs&j=146c3188-c0b9-5851-2ce6-f639204f021b&t=dab87b97-7534-55e9-dd66-48c338387810

**Comment by heoelri** [comment]
Deploying another instance to figure out why the load test results are outside of our baseline here: https://dev.azure.com/alwayson/Foundational-Online/_build/results?buildId=447&view=results

This is not related and NOT blocking this PR though. ;)

File: .ado/pipelines/config/configuration.yaml
Comment (sebader) [inline .ado/pipelines/config/configuration.yaml:45]:
```suggestion
# Embedded load testing configuration
```

So we wont forget to remove it when we move to ALT ;)

File: .ado/pipelines/config/configuration.yaml
Comment (sebader) [inline .ado/pipelines/config/configuration.yaml:47]:
```suggestion
- name: 'loadTestNumberOfWorkerNodes' # number of worker nodes
```

File: .ado/pipelines/config/configuration.yaml
Comment (sebader) [inline .ado/pipelines/config/configuration.yaml:49]:
```suggestion
- name: 'loadTestDurationInSeconds' # load test duration
```

File: .ado/pipelines/config/configuration.yaml
Comment (sebader) [inline .ado/pipelines/config/configuration.yaml:51]:
```suggestion
- name: 'loadTestNumberOfUsers' # number of simulated users in the load test
```

File: .ado/pipelines/config/configuration.yaml
Comment (sebader) [inline .ado/pipelines/config/configuration.yaml:53]:
```suggestion
- name: 'loadTestUserSpawnRate' # load test spawn rate users / per second
```

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (sebader) [inline .ado/pipelines/templates/stages-full-release.yaml:313]: you can remove the comments here

**Review by sebader — CHANGES_REQUESTED** [review]

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:313]:
```suggestion
      numberOfWorkerNodes: $(loadTestNumberOfWorkerNodes)
```

File: .ado/pipelines/config/configuration.yaml
Comment (heoelri) [reply .ado/pipelines/config/configuration.yaml:45]: no sure if these parameters are still applicable then but ok for me

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (sebader) [inline .ado/pipelines/templates/stages-full-release.yaml:314]:
```suggestion
      testDurationSeconds: $(loadTestDurationInSeconds)
```

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (sebader) [inline .ado/pipelines/templates/stages-full-release.yaml:315]:
```suggestion
      locustUser: $(loadTestNumberOfUsers)
```

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (sebader) [inline .ado/pipelines/templates/stages-full-release.yaml:316]:
```suggestion
      locustSpawnRate: $(loadTestUserSpawnRate)
```

**Review by sebader — CHANGES_REQUESTED** [review]

File: .ado/pipelines/templates/stages-full-release.yaml
Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: why do you want to drop the comments here?

Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: because they are just redundant

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: not within this file - we have comments for containerImageName, containerImageDockerFile and many other parameters in this file as well. not sure if i'd drop them here. i still think that using comments here adds value

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: or do you think the parameter name itself is self-explaining?

Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: yeah I'd say so. I mean, feel free to leave them here. Its just another set of comments that might drift

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:314]: no, i agree - i think the variable names are self-explaining

**Review by sebader — APPROVED** [review]

**Comment by heoelri** [comment]
Thank you @sebader - i made another change after you've approved the PR to use a consistent naming across all files. I'll run another test before I merge these changes into main to make sure that everything is still working.

**Comment by heoelri** [comment]
Ok, works. https://dev.azure.com/alwayson/Foundational-Online/_build/results?buildId=453&view=logs&j=146c3188-c0b9-5851-2ce6-f639204f021b&t=dab87b97-7534-55e9-dd66-48c338387810 Still issues with the load test result. But the PR itself is good to go.

**Pull request merged**
